apps/bootloader.js

```diff
@@ -101,12 +101,28 @@ if (startapp) {
   function drawWidgets() {
     for (var w of WIDGETS) w.draw();
   }
-  var clockApps = require("Storage").list().filter(a=>a[0]=='+').map(app=>{
-    try { return require("Storage").readJSON(app); }
-    catch (e) {}
-  }).filter(app=>app.type=="clock").sort((a, b) => a.sortorder - b.sortorder);
-  if (clockApps && clockApps.length > 0) eval(require("Storage").read(clockApps[0].src));
+  var settings;
+  try {
+    settings = require("Storage").readJSON('@setting');
+  } catch (e) {
+    settings = {}
+  }
+  var clockApp = settings.clock;
```

Comment:
Can we use 2 different vars, one for clock name from settings, and another for clock app object?
Something like `defaultClockApp` and `clockApp`?

Reply:
It could, but for what reason? `clockApp` is the clock app we're going to load. First we check to see if settings provides it, otherwise we fall back to the existing code to get the first clock app we find. Adding another var would only increase memory usage for little benefit.

Reply:
Point taken, didn't think about memory usage.